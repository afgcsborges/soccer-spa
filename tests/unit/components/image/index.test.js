import Image from 'components/image'
import React from 'react'
import { shallow } from 'enzyme'

const noImage =
    'https://media.istockphoto.com/vectors/image-unavailable-icon-vector-id1206575314?k=20&m=1206575314&s=170667a&w=0&h=ioRmz6_fwuW0bETWDp9JZIZdv-YayKXTdtGmjgB5pnI='

describe('Image component tests', () => {
    it('should have no image src', () => {
        const wrapper = shallow(<Image />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('img').props().src).toBe(noImage)
        expect(wrapper.find('img').props().width).toBe(150)
    })

    it('should have correct src and width', () => {
        const wrapper = shallow(<Image src={'SOME_SRC'} width={200} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('img').props().src).toBe('SOME_SRC')
        expect(wrapper.find('img').props().width).toBe(200)
    })
})
