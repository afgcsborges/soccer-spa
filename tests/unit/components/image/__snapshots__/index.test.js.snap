// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component tests should have correct src and width 1`] = `
<img
  src="SOME_SRC"
  style={
    Object {
      "float": "left",
      "marginBottom": "10px",
      "maxHeight": "100%",
      "maxWidth": "100%",
      "objectFit": "contain",
    }
  }
  width={200}
/>
`;

exports[`Image component tests should have no image src 1`] = `
<img
  src="https://media.istockphoto.com/vectors/image-unavailable-icon-vector-id1206575314?k=20&m=1206575314&s=170667a&w=0&h=ioRmz6_fwuW0bETWDp9JZIZdv-YayKXTdtGmjgB5pnI="
  style={
    Object {
      "float": "left",
      "marginBottom": "10px",
      "maxHeight": "100%",
      "maxWidth": "100%",
      "objectFit": "contain",
    }
  }
  width={150}
/>
`;
